standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5591,15 +5586,12 @@ private boolean needCheckPartitionLimit(int requestSize) {
     return partitionLimit > -1 && (requestSize < 0 || requestSize > partitionLimit);
   }
 
-  private void checkLimitNumberOfPartitions(String tblName, int numPartitions, int maxToFetch) throws MetaException {
-    if (isPartitionLimitEnabled()) {
-      int partitionLimit = MetastoreConf.getIntVar(conf, ConfVars.LIMIT_PARTITION_REQUEST);
-      int partitionRequest = (maxToFetch < 0) ? numPartitions : maxToFetch;
-      if (partitionRequest > partitionLimit) {
-        String configName = ConfVars.LIMIT_PARTITION_REQUEST.toString();
-        throw new MetaException(String.format(PARTITION_NUMBER_EXCEED_LIMIT_MSG, partitionRequest,
-            tblName, partitionLimit, configName));
-      }
+  private void checkLimitNumberOfPartitions(String tblName, int numPartitions) throws MetaException {
+    int partitionLimit = MetastoreConf.getIntVar(conf, ConfVars.LIMIT_PARTITION_REQUEST);
+    if (partitionLimit > -1 && numPartitions > partitionLimit) {
```

Comment:
" > 0" ??

Reply:
Here we followed the existing code in `isPartitionLimitEnabled()`, do we need change it?